### ISSUE TITLE
Garbage in console output (Windows 8.1)

### ISSUE THREAD

**Issue by Divetoxx** [mpv-player/mpv#2743]
![grbg](https://cloud.githubusercontent.com/assets/1689751/12468992/4d8b7174-bff9-11e5-866f-1fa3569bb1c3.png)

There is a little garbage in console output in Windows - could you see attached screenshot.
I see such four random symbols after file information.

**Comment by lachs0r** [comment]
Are you sure that’s not part of the file’s metadata?

**Comment by rossy** [comment]
I think I've seen this issue before, but for some reason I can't reproduce it at the moment. If it only started happening recently, it may be to do with recent-ish changes in how stdio is handled, such as 30bbcd6 or c19f634.

**Comment by Frechdachs** [comment]
I noticed something like this too while using libmpv (lachs0r's 0.15.0 binaries) with jaseg's python interface.
For me it happens only when angle is first initialized. (e.g. also with `es=no`, but not with `backend=win`)

I have not been able to reproduce this with lachs0r's mpv.exe.

**Comment by Frechdachs** [comment]
Oh, I did manage to reproduce it with lachs0r's mpv.exe now. It only happens with the 32 bit build. (So 64 bit libmpv probably doesn't have this problem, too.)

**Comment by rossy** [comment]
Ah, thanks. That narrows it down. I can reproduce it consistently now.

```
C:\ProgramData\mpv\32>mpv --no-config --vo=opengl:backend=angle --idle --force-window
`‗Ä
Exiting... (Quit)

C:\ProgramData\mpv\32>mpv --no-config --vo=opengl:backend=angle --idle --force-window
▄÷ 
Exiting... (Quit)

C:\ProgramData\mpv\32>mpv --no-config --vo=opengl:backend=angle --idle --force-window
8¶B
Exiting... (Quit)

C:\ProgramData\mpv\32>mpv --no-config --vo=opengl:backend=win --idle --force-window

Exiting... (Quit)

C:\ProgramData\mpv\32>
```

Seems like it always prints four characters of random garbage to stdout when initializing 32-bit ANGLE. This happens on the Windows console, in mintty and when stdout is redirected to a file. It does not appear in the output of --log-file.

Verbose log:
```
C:\ProgramData\mpv\32>mpv --no-config --vo=opengl:backend=angle --idle --force-window --msg-level=vo=v
[vo/opengl] Setting option 'backend' = 'angle' (flags = 0)
[vo/opengl] Initializing OpenGL backend 'angle'
└‗▄[vo/opengl] GL_VERSION='OpenGL ES 3.0 (ANGLE 2.1.0.)'
[vo/opengl] Detected GLES 3.0.
...
```

**Comment by Divetoxx** [comment]
If playing any video-file I see it. Without or with any mpv config too.

**Comment by rossy** [comment]
Ok, so I think I figured this out. ANGLE calls sprintf_s here:
https://github.com/google/angle/blob/618bebc/src/libANGLE/renderer/d3d/RendererD3D.cpp#L602

This calls mingw-w64's built-in sprintf_s stub, which calls vsprintf_s here:
https://github.com/Alexpux/mingw-w64/blob/5d79dd7/mingw-w64-crt/secapi/sprintf_s.c#L18

...which tries to dynamically load and call msvcrt.dll's **vprintf_s**:
https://github.com/Alexpux/mingw-w64/blob/e0ed54c/mingw-w64-crt/secapi/vsprintf_s.c#L22

The uninitialized destination buffer becomes vprintf_s' format argument, so garbage from the stack gets printed to stdout.

I'll try to fix this on my machine, then I'll report it upstream.

**Comment by rossy** [comment]
Upstream issue here (it was already reported): https://sourceforge.net/p/mingw-w64/bugs/525/

**Comment by rossy** [comment]
Fixed in mirror/mingw-w64@9975303.
